prime-router/src/main/kotlin/Schema.kt

```diff
@@ -79,6 +79,16 @@ data class Schema(
         return elementIndex[name]
     }
 
+    /**
+     * Creates a uniform trackingId based on the row number and/or the trackingElement id
+     * in the report item.
+     * @param row the zero based entry of the item in the report
+     * @param trackingId the value of the schema's trackingElement if defined
+     */
+    fun getTrackingId(row: Int, trackingId: String? = null): String {
+        return trackingId ?: "row$row"
+    }
+
```

Comment:
Had to implement a default trackingId. Without this, any schema with no tracking element returns null and I have no way to link items to outgoing reports.